soc/st/stm32/stm32u0x/Kconfig.defconfig.stm32u073xx

```
@@ -8,4 +8,24 @@ if SOC_STM32U073XX
 config NUM_IRQS
 	default 32
 
+if I2C_STM32
+
+config I2C2_I2C3_ENABLED
+	default y if $(dt_nodelabel_enabled,i2c2) && $(dt_nodelabel_enabled,i2c3)
+
+config I2C2_I2C4_ENABLED
+	default y if $(dt_nodelabel_enabled,i2c2) && $(dt_nodelabel_enabled,i2c4)
+
+config I2C3_I2C4_ENABLED
+	default y if $(dt_nodelabel_enabled,i2c3) && $(dt_nodelabel_enabled,i2c4)
+
+config I2C2_I2C3_I2C4_ENABLED
+	default y if $(dt_nodelabel_enabled,i2c2) && $(dt_nodelabel_enabled,i2c3) &&  $(dt_nodelabel_enabled,i2c4)
+
+
+config SHARED_INTERRUPTS
+	default y if I2C2_I2C3_ENABLED || I2C3_I2C4_ENABLED || I2C2_I2C4_ENABLED || I2C2_I2C3_I2C4_ENABLED
```

Could be summed up as:
```suggestion
config SHARED_INTERRUPTS
	default y if $(dt_nodelabel_enabled,i2c2) && $(dt_nodelabel_enabled,i2c3)
	default y if $(dt_nodelabel_enabled,i2c2) && $(dt_nodelabel_enabled,i2c4)
	default y if $(dt_nodelabel_enabled,i2c3) && $(dt_nodelabel_enabled,i2c4)
```

and  can be put in `soc/st/stm32/stm32u0x/Kconfig.defconfig`